syntaxes/connectors.jsonselection.yaml

```diff
@@ -0,0 +1,270 @@
+# use 'js-yaml' to convert to json
+# npx js-yaml syntaxes/connectors.jsonselection.yaml >| syntaxes/connectors.jsonselection.json
+name: "Apollo Connectors JSONSelection"
```

Comment:
Is this user-facing at all? We've avoided using this name anywhere user-facing for now, since we're going to change it, we just haven't decided to what yet 😅

Reply:
That's a good call-out - the only "user-facing" thing right now is that we introdude a new language to VSCode:

https://github.com/apollographql/vscode-graphql/blob/0ea12de909a837847cdedc704c80f4278f3ec879/package.json#L159-L165

That means it will pop up if you edit a file called `test.apollo.jsonselection`, but also if you use the "change file language" dropdown:

![screenshot of that dropdown](image.png)

We might want to give that language a better user-facing name.

It's technically also visible if you inspect syntax highlighting tokens, but I would not really consider that "user-facing":

![screenshot](image.png)

Reply:
Yeah, that sounds generic enough to be okay. We also use the term "mapping" a lot when referring to it right now, like "Apollo Connectors Mapping Syntax", though that's really verbose.

Reply:
![image](https://github.com/user-attachments/assets/74f57c64-12a0-4858-b394-56363ec5318a)

Renamed it, and also put it into `[]` because that will sort it alphabetically at the bottom of the list. There's no need for this implementation detail to be the first thing on the list every time people open it :)